pkg/apis/core/types.go

```diff
@@ -3504,6 +3506,21 @@ type ServiceSpec struct {
 	// cluster (e.g. IPv6 in IPv4 only cluster) is an error condition and will fail during clusterIP assignment.
 	// +optional
 	IPFamily *IPFamily
+
+	// topologyKeys is a preference-order list of topology keys which
+	// implementations of services should use to preferentially sort endpoints
+	// when accessing this Service. Topology keys must be valid label keys and
+	// at most 16 keys may be specified, and can't be used with externalTrafficPolicy=Local.
+	// Endpoints are chosen based on the first topology key with available backends,
+	// starting at index 0.
```

Comment:
I don't think we need the "starting with" clause.

Reply:
We might want to think about the wording of this past alpha.  If someone built a really smart, load-based balancer, and they could detect that "same zone" endpoints were overloaded, but "nearby zone" was OK - do we think that would be allowable.  Initially I thought not, but the more I think about it, the more I think it should probably be allowed.

no change needed for now, but think about it.